vscode/src/telemetry/impl/postTelemetry.ts

```diff
@@ -0,0 +1,86 @@
+/*
+  Copyright (c) 2024, Oracle and/or its affiliates.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+import { LOGGER } from "../../logger";
+import { TELEMETRY_API } from "../config";
+import { BaseEvent } from "../events/baseEvent";
+
+interface TelemetryEventResponse {
+    statusCode: number;
+    event: BaseEvent<any>;
+};
+
+export interface TelemetryPostResponse {
+    success: TelemetryEventResponse[];
+    failures: TelemetryEventResponse[];
+};
+
+export class PostTelemetry {
+    public post = async (events: BaseEvent<any>[]): Promise<TelemetryPostResponse> => {
+        try {
+            if (TELEMETRY_API.baseEndpoint == null) {
```

Comment:
Perhaps this is supposed to be:
```suggestion
            if (TELEMETRY_API.baseUrl == null) {
```